tools/clang/test/HLSLFileCheck/passes/llvm/instcombine/sink.hlsl

```diff
@@ -0,0 +1,38 @@
+// RUN: %dxc %s /T ps_6_0 -opt-disable sink | FileCheck %s
+
+// Make sure the selects are NOT sunk into a single block
+
+cbuffer cb {
+    bool cond[6];
+    bool br[6];
+    float a[6];
+}
+
+float main() : SV_Target {
+    float val = 0;
+    float ret = 0;
+    // CHECK: br
+    if (br[0]) {
+        // CHECK: select
```

Comment:
Just to confirm, the change to instcombine is not restricted to selects, right? It will stop sinking of all instruction types?

Reply:
That's correct. It stops sinking everything.